zzzz_modular_occulus/code/game/gamemodes/events/siren_scan.dm

```diff
@@ -0,0 +1,100 @@
+	spawnLists += list(list(/mob/living/simple_animal/hostile/siren/augmentor, /mob/living/simple_animal/hostile/siren/augmentor, /mob/living/simple_animal/hostile/siren/replicant, /mob/living/simple_animal/hostile/siren/replicant))
+	spawnLists += list(list(/mob/living/simple_animal/hostile/siren/composer, /mob/living/simple_animal/hostile/siren/replicant))
+	spawnLists += list(list(/mob/living/simple_animal/hostile/siren/conservator,/mob/living/simple_animal/hostile/siren/conservator, /mob/living/simple_animal/hostile/siren/augmentor, /mob/living/simple_animal/hostile/siren/composer))
+
+/datum/event/siren_scan/start()
+	SSweather.run_weather(/datum/weather/siren_scan)
+	runThisOnEventStartup()
+
+/datum/event/siren_scan/tick()
+	if(activeFor == enterBelt)
+		command_announcement.Announce("The Anomalous electromagnetic interference has reached the ship. Unable to determine source.", "Anomaly Alert", new_sound = 'zzzz_modular_occulus/sound/effects/The_Siren.wav')
+		scanning()
+		for(var/datum/weather/siren_scan/R in SSweather.processing)
+			R.start()
+
+	if(activeFor >= enterBelt && activeFor <= leaveBelt)
+		postStartTicks++
+
+	if(postStartTicks == scanInterval)
+		postStartTicks = 0
+		scanning()
+
+	else if(activeFor == leaveBelt)
+		for(var/datum/weather/siren_scan/R in SSweather.processing)
+			R.wind_down()
+		command_announcement.Announce("The Anomalous electromagnetic interference has ceased. Engineering inspections reccomended.", "Anomaly Alert", new_sound = 'zzzz_modular_occulus/sound/effects/The_Siren.wav')
+
+
+
+/datum/event/siren_scan/proc/PickSirenPod(var/mob/candidate)
+	var/list/spawnTypes = pick_n_take(spawnLists)
+	SpawnSirenPodInRange(candidate,10,7,spawnTypes)
+
+/datum/event/siren_scan/proc/SpawnSirenPodInRange(atom/origin,outer_range,inner_range,list/spawnTypes)
+	for(var/type in spawnTypes)
+		var/turf/picked = get_random_secure_turf_in_range(origin,10,7)
+		type = pick(spawnTypes)
+		new type(picked)
+
+/datum/event/siren_scan/proc/scanning()	//this proc will eventually be made to locate any players, and spawn a cluster of siren mobs out of view on the ship. Not needed for the event tho.
+
+
+	var/list/candidates = list()	//list of candidate keys
+	for(var/mob/living/carbon/human/G in GLOB.player_list)
+		if(G.mind && G.stat != DEAD && G.is_client_active(5) && !player_is_antag(G.mind))
+			if(isOnStationLevel(G))
+				candidates += G
+	if(!candidates.len)
+		return
+	candidates -= used_candidates
+	candidates = shuffle(candidates)//Incorporating Donkie's list shuffle
+
+	var/siren_anger = rand(2, 4)
+	while(siren_anger > 0 && candidates.len)
+		PickSirenPod(candidates[1])
+		used_candidates += candidates[1]
+		candidates.Remove(candidates[1])
```

Comment:
shuffle() is much more expensive than pick(), as the former creates a new list.
Use pick(), and don't forget to change the stuff in here.